consensus/src/dag/dag_store.rs

```diff
@@ -0,0 +1,268 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, ensure};
+use aptos_consensus_types::common::{Author, Payload, Round};
+use aptos_crypto::{
+    hash::{CryptoHash, CryptoHasher},
+    HashValue,
+};
+use aptos_crypto_derive::CryptoHasher;
+use aptos_types::{aggregate_signature::AggregateSignature, validator_verifier::ValidatorVerifier};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{BTreeMap, HashMap, HashSet},
+    ops::Deref,
+    sync::Arc,
+};
+
+/// Represents the metadata about the node, without payload and parents from Node
+#[derive(Clone, Serialize, Deserialize)]
+pub struct NodeMetadata {
+    epoch: u64,
+    round: Round,
+    author: Author,
+    timestamp: u64,
+    digest: HashValue,
+}
+
+/// Node representation in the DAG, parents contain 2f+1 strong links plus weak links
```

Comment:
Would be useful to explain what strong/weak links are.

Reply:
will add more comments after the code gets more mature